tests/test_app/test_composable.py

```diff
@@ -923,5 +923,20 @@ def main(self, val: int) -> int:
     __app_registry.pop(get_object_provenance(app_non_composable2), None)
 
 
+def test_handles_numpy_array_types():
+    """apps correctly handle numpy return types"""
+    from numpy import array, ndarray
+
+    @define_app
+    def power(val: ndarray, pow: int) -> int:
+        return val ** pow
+
+    sqd = power(pow=2)
+    d = array([3, 3])
+    assert (sqd(d) == array([9, 9])).all()
+
+    __app_registry.pop(get_object_provenance(power), None)
```

Comment:
I think we need tests for non-numpy arrays and other cases of error in _call function.

Reply:
Good point. I do wonder how general this should be. For example, if a function returns `0`, this will raise an exception. But should it? This makes make me think our definition of a null result is too broad.

Perhaps we should limit our error definition to the case where `result is None` and let everything else through?

Reply:
Yes, as you said, when return value is an empty list or zero value, it should be ok.
What if val (input value) was an empty list?